### PR TITLE
Adds support for server middlewares

### LINKED ISSUE #11: Support applying server middlewares
We need to be able to import the server instance to apply middlewares on it.

For example, in `src/config.js` adding following:

```js
import { server } from 'graphpack';

server.applyMiddleware(/* ... */);
```

More details about applying middlewares in the [Apollo Server docs](https://www.apollographql.com/docs/apollo-server/api/apollo-server.html#ApolloServer-applyMiddleware).

### LINKED ISSUE #14: Integration with existing Express app ?
Is it possible to integrate the package with an existing express.js application ?

### PR DESCRIPTION
**Pull request by farskid**
Adds support for applyMiddleware.

closes #11 & closes #14

### CODE REVIEW

File: packages/graphpack/lib/server.js [hunk 1]
```diff
@@ -28,6 +29,9 @@ const server = new ApolloServer({
   resolvers,
 });
 
+// Apply user provided middlewares
+server.applyMiddleware(applyMiddleware);
```
Comment (glennreyes) [inline packages/graphpack/lib/server.js]: Maybe let's only call this function only if there are valid middlewares?

Comment (farskid) [reply packages/graphpack/lib/server.js]: I'm ok with doing it if there is any middlewares available. could you please elaborate more on validating?
According to apollo-server-[integration] libs, the only required parameter for a middleware is app which should be an instance of that server type. is checking for `applyMiddleware.app` enough?

Comment (glennreyes) [reply packages/graphpack/lib/server.js]: Yep that should be okay! I’m also thinking of moving the entire config.js into backpack.config.js but that would be another PR/issue.

Comment (farskid) [reply packages/graphpack/lib/server.js]: That we can do on another PR :)
QQ: How would you pass multiple middlewares using this pattern? is it already handled in Apollo server or we should accept a list of middlewares?

Comment (glennreyes) [reply packages/graphpack/lib/server.js]: For that I’d stick with how Apollo is handling it, so passing the app that has all the middlewares applied.

I’ll aim to merge this once github is fixed 👍

**Comment by farskid** [comment]
Thanks Glen. Now it only adds middlewares if any valid one is passed.

**Pull request merged**